src/components/file-upload/FileUpload.tsx

```diff
@@ -7,52 +7,88 @@ import {
 } from 'baseui/file-uploader';
 import FileUploadInstruction from './FileUploadInstruction';
 import FileUploadPreview from './FileUploadPreview';
-import { padding } from '../../utils';
-
-const fileUploaderOverrides = (
-  instructionMessage: string,
-  preview?: string,
-): BaseFileUploaderOverrides<StyleProps> => ({
-  Root: {
-    style: {
-      height: '100%',
-    },
-  },
-  FileDragAndDrop: {
-    style: {
-      margin: '0',
-      ...padding('0'),
-      display: 'flex',
-      alignContent: 'center',
-      justifyContent: 'center',
-      height: '100%',
-    },
-  },
-  ButtonComponent: {
-    props: {
-      size: 'compact',
-      kind: 'secondary',
-      shape: 'pill',
-    },
-  },
-  ContentMessage: {
-    // eslint-disable-next-line react/display-name
-    component: preview ? FileUploadPreview : FileUploadInstruction,
-    props: {
-      message: instructionMessage,
-      preview,
-    },
-  },
-  ContentSeparator: {
-    component: () => null,
-  },
-});
+import { borderRadius, margin, padding } from '../../utils';
+import { Button } from 'baseui/button';
+import { useTheme } from '../../providers/ThemeProvider';
 
 export interface FileUploadProps extends Omit<BaseFileUploaderProps, 'overrides'> {
   instructionMessage: string;
+  buttonText: string;
   preview?: string;
+  error?: boolean;
 }
 
-export const FileUpload = ({ instructionMessage, preview, ...rest }: FileUploadProps) => {
-  return <BaseFileUploader {...rest} overrides={fileUploaderOverrides(instructionMessage, preview)} />;
+export const FileUpload = ({ instructionMessage, preview, buttonText, error, ...rest }: FileUploadProps) => {
+  const { theme } = useTheme();
+  const {
+    current: {
+      sizing: { scale100 },
+      colors: { negative },
+      customColors: { dark0 },
+    },
+  } = theme;
+
+  const fileUploaderOverrides = (
+    instructionMessage: string,
+    buttonText: string,
+    preview?: string,
+    error?: boolean,
+  ): BaseFileUploaderOverrides<StyleProps> => ({
+    Root: {
+      style: {
+        height: '100%',
+        background: 'red',
```

Comment:
@Ihor-Lavrov should it be just `red`, not any value from theme?

Reply:
fixed